trace-sdk/src/main/java/io/bitrise/trace/network/MetricSender.java

```diff
@@ -92,6 +96,39 @@ public void onSuccess() {
         });
     }
 
+    @VisibleForTesting
+    protected static int headerComparison(@NonNull final MetricRequest request,
```

Comment:
Please consider splitting this method to smaller parts, to make it more readable and testable.

Is it required to make it protected? If not, I would recommend to make it more restrictive and use the package private visibility.

Javadoc is missing, but it would help developers to understand what it does without checking the content.

nit: when I read the name, it confuses a bit me, because I would expect what it does (verb instead of a noun). Could you please make this more descriptive, something like `countNonAcceptedMetrics` or something like that?

Reply:
Thanks for the feedback!

I moved the logic to generate the list of keys to the MetricUtils (and wrote tests) - hopefully that makes it clearer to see what's happening now?

Updated to make it package private, added javadoc, and renamed to `countHeaderComparisonDifference` - i think the difference is better than "non accepted" because there is the scenario where the backend accepts more than we think we sent. I added a test for this case too. Does that make sense?

Reply:
Thanks for the updates.